packages/protobuf/src/types.ts

```diff
@@ -89,6 +94,65 @@ export type UnknownField = {
   readonly data: Uint8Array;
 };
 
+/**
+ * Describes a streaming RPC declaration.
+ */
+export type DescMethodStreaming<
+  I extends DescMessage = DescMessage,
+  O extends DescMessage = DescMessage,
+> =
+  | DescMethodClientStreaming<I, O>
+  | DescMethodServerStreaming<I, O>
+  | DescMethodBiDiStreaming<I, O>;
+
+/**
+ * Describes a unary RPC declaration.
+ */
+export type DescMethodUnary<
+  I extends DescMessage = DescMessage,
+  O extends DescMessage = DescMessage,
+> = DescMethodCommon & {
+  methodKind: "unary";
+  input: I;
+  output: O;
```

Comment:
The corresponding `DescMethod` properties are readonly. I think it would be good to make the replacements readonly as well.